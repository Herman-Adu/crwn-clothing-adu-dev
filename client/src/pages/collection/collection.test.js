import React from 'react';
import { shallow } from 'enzyme';

import { CollectionPage } from './collection.component';
import CollectionItem from '../../components/collection-item/collection-item.component';

describe('collectionPage', () => {
    
    let wrapper; 
    let mockItems = [{id: 1}, {id: 2}, {id: 3}];

    // setup collection
    beforeEach(() => {
        const mockCollection = {
            items: mockItems,
            title: 'Test Collection'
        };

        wrapper = shallow(<CollectionPage collection={mockCollection} />)
    });

    it('should render the CollectionPage component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the same number of CollectionItems as collection array', () => {
        expect(wrapper.find(CollectionItem).length).toBe(mockItems.length);
    });
});

