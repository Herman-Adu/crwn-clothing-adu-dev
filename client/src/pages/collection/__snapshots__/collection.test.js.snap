// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collectionPage should render the CollectionPage component 1`] = `ShallowWrapper {}`;
